### PR TITLE
DBZ-7758 Add ehcache transaction cache implementation

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-7758

Supersedes #5485 

- [x] Consider configuration as supplied XML options, like Infinispan
- [x] Revert forcing ehcache in test suite
- [x] A few refactors wrt new caching API added in DBZ-8054.

### CODE REVIEW

**Comment by som-design** [comment]
I apologize, right now I don't have much free time so I have to take care
of sick people.

ในวันที่ อ. 6 ส.ค. 2024 05:53 น. Chris Cranford ***@***.***>
เขียนว่า:

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/ehcache/serialization/AbstractSerializerStream.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public abstract class AbstractSerializerStream implements AutoCloseable {
+    /**
+     * Arrays cannot be serialized with null values and so we use a sentinel value
+     * to mark a null element in an array.
+     */
+    protected static final String NULL_VALUE_SENTINEL = "$$DBZ-NULL$$";
+
+    /**
+     * The supplied value arrays can now be populated with {@link OracleValueConverters#UNAVAILABLE_VALUE}
+     * which is simple java object.  This cannot be represented as a string in the cached Infinispan record
```
[the remark this answers is not in the document]
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/ehcache/serialization/AbstractSerializerStream.java:22]: Great catch, cut-n-paste got me :)

**Comment by jpechane** [comment]
@Naros LGTM, thanks! We just need to talk about testing of different caches, but that's something that can wait for the next week.

**Pull request merged**